Examples/Scripts/TrackingPerformance/boundParamResolution.C

```diff
@@ -0,0 +1,852 @@
+  tree->SetBranchAddress("res_eLOC0_smt", &res_LOC0_smt);
+  tree->SetBranchAddress("res_eLOC1_smt", &res_LOC1_smt);
+  tree->SetBranchAddress("res_ePHI_smt", &res_PHI_smt);
+  tree->SetBranchAddress("res_eTHETA_smt", &res_THETA_smt);
+  tree->SetBranchAddress("res_eQOP_smt", &res_QOP_smt);
+  tree->SetBranchAddress("res_eT_smt", &res_T_smt);
+
+  tree->SetBranchAddress("pull_eLOC0_prt", &pull_LOC0_prt);
+  tree->SetBranchAddress("pull_eLOC1_prt", &pull_LOC1_prt);
+  tree->SetBranchAddress("pull_ePHI_prt", &pull_PHI_prt);
+  tree->SetBranchAddress("pull_eTHETA_prt", &pull_THETA_prt);
+  tree->SetBranchAddress("pull_eQOP_prt", &pull_QOP_prt);
+  tree->SetBranchAddress("pull_eT_prt", &pull_T_prt);
+  tree->SetBranchAddress("pull_eLOC0_flt", &pull_LOC0_flt);
+  tree->SetBranchAddress("pull_eLOC1_flt", &pull_LOC1_flt);
+  tree->SetBranchAddress("pull_ePHI_flt", &pull_PHI_flt);
+  tree->SetBranchAddress("pull_eTHETA_flt", &pull_THETA_flt);
+  tree->SetBranchAddress("pull_eQOP_flt", &pull_QOP_flt);
+  tree->SetBranchAddress("pull_eT_flt", &pull_T_flt);
+  tree->SetBranchAddress("pull_eLOC0_smt", &pull_LOC0_smt);
+  tree->SetBranchAddress("pull_eLOC1_smt", &pull_LOC1_smt);
+  tree->SetBranchAddress("pull_ePHI_smt", &pull_PHI_smt);
+  tree->SetBranchAddress("pull_eTHETA_smt", &pull_THETA_smt);
+  tree->SetBranchAddress("pull_eQOP_smt", &pull_QOP_smt);
+  tree->SetBranchAddress("pull_eT_smt", &pull_T_smt);
+
+  tree->SetBranchAddress("g_x_prt", &g_x_prt);
+  tree->SetBranchAddress("g_y_prt", &g_y_prt);
+  tree->SetBranchAddress("g_z_prt", &g_z_prt);
+  tree->SetBranchAddress("g_x_flt", &g_x_flt);
+  tree->SetBranchAddress("g_y_flt", &g_y_flt);
+  tree->SetBranchAddress("g_z_flt", &g_z_flt);
+  tree->SetBranchAddress("g_x_smt", &g_x_smt);
+  tree->SetBranchAddress("g_y_smt", &g_y_smt);
+  tree->SetBranchAddress("g_z_smt", &g_z_smt);
+
+  tree->SetBranchAddress("nStates", &nStates);
+  tree->SetBranchAddress("nMeasurements", &nMeasurements);
+  tree->SetBranchAddress("volume_id", &volume_id);
+  tree->SetBranchAddress("layer_id", &layer_id);
+  tree->SetBranchAddress("module_id", &module_id);
+  tree->SetBranchAddress("predicted", &predicted);
+  tree->SetBranchAddress("filtered", &filtered);
+  tree->SetBranchAddress("smoothed", &smoothed);
+
+  // Section 3: Histogram booking
+  TCanvas* rangeCanvas =
+      new TCanvas("rangeCanvas", "Range Estimation", 10, 10, 900, 600);
+  rangeCanvas->Divide(3, 2);
+  std::vector<std::string> res_smts = {"res_eLOC0_smt", "res_eLOC1_smt",
+                                       "res_ePHI_smt",  "res_eTHETA_smt",
+                                       "res_eQOP_smt",  "res_eT_smt"};
+
+  /// Helper method to make a volLayId string for identification & cut
+  ///
+  /// ensures a unique identification
+  auto volLayIdCut = [](int vol, int lay) -> std::array<std::string, 2> {
+    if (vol < 0 and lay < 0) {
+      return {std::string("all_"), ""};
+    }
+
+    std::string vlstr = "vol" + std::to_string(vol);
```

Comment:
It seems like we should have the volume id cut only if `vol > 0`, and have the layer id cut only if `vol > 0 and layer > 0`, e.g.  an input of `vol =  -1` and `lay = 8` can mean that no cut is necessary ?

Reply:
if `volume_id < 0` then necessarily `layer_id < 0`, hence this case is covered by

```c++
    if (vol < 0 and lay < 0) {
      return {std::string("all_"), ""};
    }
```

In L366.